### PR TITLE
OCT-1942 Projects search: client

### PR DESCRIPTION
**Pull request by aziolek**
## Definition of Done

1. [ ] If required, the desciption of your change is added to the [QA changelog](https://www.notion.so/octantapp/Changelog-for-the-QA-d96fa3b411cf488bb1d8d9a598d88281) 
2. [ ] Acceptance criteria are met.
3. [ ] PR is manually tested before the merge by developer(s).
    - [ ] Happy path is manually checked.
4. [ ] PR is manually tested by QA when their assistance is required (1).
    - [ ] Octant Areas & Test Cases are checked for impact and updated if required (2).
5. [ ] Unit tests are added unless there is a reason to omit them.
6. [ ] Automated tests are added when required.
7. [ ] The code is merged.
8. [ ] Tech documentation is added / updated, reviewed and approved (including mandatory approval by a code owner, should such exist for changed files).
    - [ ] BE: Swagger documentation is updated.
9. [ ] When required by QA:
    - [ ] Deployed to the relevant environment.
    - [ ] Passed system tests.

---

(1) Developer(s) in coordination with QA decide whether it's required. For small tickets introducing small changes QA assistance is most probably not required.

(2) [Octant Areas & Test Cases](https://docs.google.com/spreadsheets/d/1cRe6dxuKJV3a4ZskAwWEPvrFkQm6rEfyUCYwLTYw_Cc).

### CODE REVIEW

File: client/src/views/ProjectsView/ProjectsView.tsx [hunk 1]
```diff
@@ -108,6 +187,9 @@ const ProjectsView = (): ReactElement => {
 
   return (
     <>
+      <ViewTitle className={styles.viewTitle}>
+        {t('viewTitle', { epochNumber: currentEpoch })}
```
Comment (jmikolajczyk) [inline client/src/views/ProjectsView/ProjectsView.tsx:191]: When AW is open or AW is closed and `areCurrentEpochsProjectsHiddenOutsideAllocationWindow` is true we need to show `currentEpoch -1`.

File: client/src/hooks/queries/useSearchedProjectsDetails.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    return {
+      data: [],
+      isFetching: isFetchingQueries,
+      refetch,
+    };
+  }
+
+  return {
+    data: queries.map(({ data }) => {
+      const rewards = data && data[1] ? data[1].rewards : [];
```
Comment (jmikolajczyk) [inline client/src/hooks/queries/useSearchedProjectsDetails.ts:72]:
```suggestion
      const rewards =  data?.[1] ?? [];
```

File: client/src/hooks/queries/useSearchedProjectsDetails.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      const address = data && data[4] ? data[4] : undefined;
```
Comment (jmikolajczyk) [inline client/src/hooks/queries/useSearchedProjectsDetails.ts:73]:
```suggestion
      const address = data?.[4]
```

File: client/src/hooks/queries/useSearchedProjectsDetails.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      const rewardsOfProject = rewards.find(element => element.address === address);
+      const rewardsOfProjectMatched = rewardsOfProject
+        ? parseUnitsBigInt(rewardsOfProject.matched, 'wei')
+        : BigInt(0);
+      const rewardsOfProjectAllocated = rewardsOfProject
+        ? parseUnitsBigInt(rewardsOfProject.allocated, 'wei')
+        : BigInt(0);
+
+      return {
+        address: data && data[4] ? data[4] : undefined,
```
Comment (jmikolajczyk) [inline client/src/hooks/queries/useSearchedProjectsDetails.ts:83]:
```suggestion
        address,
```

File: client/src/hooks/queries/useSearchedProjectsDetails.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        donations: rewardsOfProjectAllocated,
+        epoch: data && data[3] ? data[3] : undefined,
```
Comment (jmikolajczyk) [inline client/src/hooks/queries/useSearchedProjectsDetails.ts:85]:
```suggestion
        epoch: data?.[3],
```

File: client/src/hooks/queries/useSearchedProjectsDetails.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        matchedRewards: rewardsOfProjectMatched,
+        numberOfDonors: data && data[2] ? data[2].length : 0,
```
Comment (jmikolajczyk) [inline client/src/hooks/queries/useSearchedProjectsDetails.ts:87]:
```suggestion
        numberOfDonors: data?.[2].length ?? 0,
```

File: client/src/hooks/queries/useSearchedProjectsDetails.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        totalValueOfAllocations: rewardsOfProjectMatched + rewardsOfProjectAllocated,
+        ...(data && data[0] ? data[0] : {}),
```
Comment (jmikolajczyk) [inline client/src/hooks/queries/useSearchedProjectsDetails.ts:89]:
```suggestion
        ...(data?.[0] ?? {}),
```

**Review by jmikolajczyk — CHANGES_REQUESTED** [review]

**Comment by jmikolajczyk** [comment]
After typing `searchPhrase`, the first render shows an empty state message and then loading state.

File: client/src/components/Projects/ProjectsList/ProjectsList.module.scss [hunk 8]
```diff
@@ -1,16 +1,4 @@
...
-  flex: 1;
-  color: $color-octant-grey5;
-  margin-bottom: 1.6rem;
-
-  .image {
-    width: 28rem;
-    margin-bottom: 3.2rem;
-  }
-}
+$elementMargin: 1.6rem;
```
[the remark this answers is not in the document]
Comment (aziolek) [reply client/src/components/Projects/ProjectsList/ProjectsList.module.scss:1]: Done.

File: client/src/hooks/queries/useSearchedProjectsDetails.ts (hunk 2 above)
Comment (aziolek) [reply client/src/hooks/queries/useSearchedProjectsDetails.ts:72]: Done.

File: client/src/hooks/queries/useSearchedProjectsDetails.ts (hunk 3 above)
Comment (aziolek) [reply client/src/hooks/queries/useSearchedProjectsDetails.ts:73]: Done.

File: client/src/hooks/queries/useSearchedProjectsDetails.ts (hunk 4 above)
Comment (aziolek) [reply client/src/hooks/queries/useSearchedProjectsDetails.ts]: Done.

File: client/src/hooks/queries/useSearchedProjectsDetails.ts (hunk 5 above)
Comment (aziolek) [reply client/src/hooks/queries/useSearchedProjectsDetails.ts]: Done.

File: client/src/hooks/queries/useSearchedProjectsDetails.ts (hunk 6 above)
Comment (aziolek) [reply client/src/hooks/queries/useSearchedProjectsDetails.ts]: Done.

File: client/src/hooks/queries/useSearchedProjectsDetails.ts (hunk 7 above)
Comment (aziolek) [reply client/src/hooks/queries/useSearchedProjectsDetails.ts]: Done.

**Comment by aziolek** [comment]
> After typing `searchPhrase`, the first render shows an empty state message and then loading state.

Done.

**Review by jmikolajczyk — APPROVED** [review]

**Pull request merged**